src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -98,29 +99,56 @@ class WorkflowView extends Component {
     const { workspaceNamespace, workspaceName, workflowNamespace, workflowName } = this.props
     const workspace = Rawls.workspace(workspaceNamespace, workspaceName)
 
-    workspace.entities().then(entities =>
-      this.setState({
-        entityTypes: _.map(e => ({ value: e, label: e.replace('_', ' ') }), _.keys(entities))
-      }))
+    const entityTypes = _.map(
+      e => ({ value: e, label: e.replace('_', ' ') }),
+      _.keys(await workspace.entities())
+    )
 
     const { invalidInputs, invalidOutputs, methodConfiguration: config } =
       await workspace.methodConfig(workflowNamespace, workflowName).validate()
-    this.setState({ invalid: { inputs: invalidInputs, outputs: invalidOutputs }, config })
 
     const processIO = _.flow(
       _.update('inputs', preprocessIOList),
       _.update('outputs', preprocessIOList)
     )
 
-    this.setState({ inputsOutputs: processIO(await Rawls.methodConfigInputsOutputs(config)) })
+    const processedIO = processIO(await Rawls.methodConfigInputsOutputs(config))
+
+    this.setState({
+      inputsOutputs: processedIO,
+      ...this.createInvalidIOMap(invalidInputs, invalidOutputs, config, processedIO),
+      config, entityTypes
+    })
+  }
+
+  createInvalidIOMap = (invalidInputs, invalidOutputs, config, io = this.state.inputsOutputs) => {
+    const findMissing = ioKey => _.flow(
+      _.reject('optional'),
+      _.filter(({ name }) => !config[ioKey][name]),
+      _.map(({ name }) => ({ [name]: 'This attribute is required' })),
+      _.mergeAll
+    )
+
+    return {
+      invalid: {
+        inputs: _.merge(invalidInputs, findMissing('inputs')(io.inputs)),
+        outputs: _.merge(invalidOutputs, findMissing('outputs')(io.outputs))
+      }
```

Comment:
The backend doesn't validate required fields for us?

Reply:
Unfortunately, no.